### PR TITLE
Adding redux

### PR DESCRIPTION
**Pull request by elberthcorniell**
In this pull request, we want to add:

- Form for books creation,
- Books list,
- Redux

### CODE REVIEW

File: src/reducers/books.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+];
+
+const booksReducer = (state = defaultBooks, action) => {
+  const { book } = action;
+  switch (action.type) {
+    case 'CREATE_BOOK':
+    default:
+      return createBook(book, state);
+    case 'REMOVE_BOOK':
+      return removeBook(book, state);
```
Comment (MuriloRoque) [inline src/reducers/books.js]: - [x] Please reformulate this function, as it's not optimal. Your `default` case should be the `state`, and inside of the `case 'CREATE_BOOK'` block, you should call `createBook` function, does it make sense?

File: src/actions/index.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  ...state,
+  book,
+];
+
+export const removeBook = (book, state) => {
+  const newState = [];
+  state.forEach(({ bookId }) => {
+    if (bookId !== book.bookId) { newState.push(book); }
+  });
+  return newState;
```
Comment (MuriloRoque) [inline src/actions/index.js]: - [x] **(OPTIONAL)** Why don't you use the function `filter`? This would make your code shorter :smile:

**Review by MuriloRoque — CHANGES_REQUESTED** [review]
## Status: Changes requested :red_circle: 

### Hello @elberthcorniell :wave: 

### You have done an amazing job in this milestone, everything is in place, there are only some minor observations I made to make your code better. Please check my comments in your code below.

Kindly make the requested changes and submit them for another review, please :pray: 

I will be on Slack 24/7 in case you have any questions about this review :handshake: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

File: src/actions/index.js (hunk 2 above)
Comment (elberthcorniell) [reply src/actions/index.js]: Yeah hehe, thank you

**Review by MuriloRoque — APPROVED** [review]
## Status: Approved :green_circle: 

### Hello again @elberthcorniell :wave: 

### Thank you for making all of the requested changes. Your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Pull request merged**